emmet-core/emmet/core/qc_tasks.py

```diff
@@ -0,0 +1,609 @@
+# mypy: ignore-errors
+
+""" Core definition of a Q-Chem Task Document """
+from typing import Any, Dict, List, Optional
+import logging
+import os
+import re
+from collections import OrderedDict
+from pydantic import BaseModel, Field
+from pymatgen.core.structure import Molecule
+from pymatgen.io.qchem.inputs import QCInput
+from monty.serialization import loadfn
+from typing import Type, TypeVar, Union
+from emmet.core.structure import MoleculeMetadata
+from pathlib import Path
+from emmet.core.qchem.calc_types import (
+    LevelOfTheory,
+    CalcType,
+    TaskType,
+)
+from emmet.core.qchem.calculation import Calculation
+
+from emmet.core.qchem.task import QChemStatus
+
+
+__author__ = (
+    "Evan Spotte-Smith <ewcspottesmith@lbl.gov>, Rishabh D. Guha <rdguha@lbl.gov>"
+)
+
+logger = logging.getLogger(__name__)
+_T = TypeVar("_T", bound="TaskDoc")
+# _DERIVATIVE_FILES = ("GRAD", "HESS")
+
+
+# class QChemStatus(ValueEnum):
+#     """
+#     Q-Chem Calculation State
+#     """
+
+#     SUCCESS = "successful"
+#     FAILED = "unsuccessful"
+
+
+class OutputDoc(BaseModel):
+    molecule: Molecule = Field(
+        None,
+        title="Output Molecule",
+        description="Output molecule after the QChem Calculation",
+    )
+
+    initial_molecule: Molecule = Field(None, description="Input Molecule object")
+    optimized_molecule: Molecule = Field(None, description="Optimized Molecule object")
+
+    # TODO: Discuss with Evan if these go here
```

Comment:
I'm not sure that they need to go in the OutputDoc specifically, but I do want them to be part of the drone/parsing/task doc creation! The build pipeline - especially the molecule association builder - is much more efficient if it can query by hashes rather than by formula alone. So it'd be good if tasks could have these from the beginning.